### DISCUSSION TITLE
Clocking into org-agenda todos with `hammy-start-org-clock-in`

### DISCUSSION THREAD

**Discussion by krisbalintona** [alphapapa/hammy.el#27] [Q&A]
Currently, `hammy-start-org-clock-in` only clocks in the org headline at point and prompts for a recent todo to clock in to otherwise. I think it would be generally desirable if `hammy-start-org-clock-in` was able to clock into tasks in the org-agenda. So instead of this:
``` emacs-lisp
(if (and (eq major-mode 'org-mode)
         (not (org-before-first-heading-p)))
    ;; At an Org entry: clock in to heading at point.
    (org-clock-in)
  ;; Not in an Org entry: offer a list to choose from.
  (org-clock-in '(4)))
```
we would have something like this:
``` emacs-lisp
(cond
 ((and (eq major-mode 'org-mode)
       (not (org-before-first-heading-p)))
  ;; At an Org entry: clock in to heading at point.
  (org-clock-in))
 ((and (eq major-mode 'org-agenda-mode)
       (org-get-at-bol 'org-marker))
  ;; At an entry in Org-agenda: clock in to todo at point.
  (org-agenda-clock-in))
 (t
  ;; Not in an Org entry: offer a list to choose from.
  (org-clock-in '(4))))
```

**Comment by alphapapa** [comment]
You should be able to do this with Embark.  Bind `hammy-start-org-clock-in` in the appropriate Embark keymap, then `embark-act` on a line in an Org Agenda or `org-ql-view` buffer, and choose `hammy-start-org-clock-in`.  Embark really opens up a whole world of possibilities this way: I use it to change to-do keywords, clock in/out, make links, and refile, from agenda/view buffers, from `org-ql-find` completion, etc, all without having to visit the entry.  @oantolin is the hero who made it possible.

**Reply by krisbalintona** [reply]
Hmmm, that's a good workflow.

However, wouldn't it still be preferable to have hammy alone suffice to clock in tasks in org-agenda buffers, without Embark? I think it's quite natural to think that if `hammy-start-org-clock-in` works on headlines it would also work in the agenda.

Or, alternatively, using Embark on org-agenda items can be mentioned in the README?
